### PR TITLE
Add helper to modify Env configs (RIPD-1247)

### PR DESCRIPTION
**Pull request by mellery451**
Add envconfig test helper for manipulating Env config via
callables. Create new common modifiers for non-admin config,
validator config and one for using different server port values.

### CODE REVIEW

File: src/test/rpc/ServerInfo_test.cpp [hunk 1]
```diff
@@ -20,53 +20,31 @@
...
-// Format manifest string to test trim()
 static auto const manifest =
-    "    JAAAAAFxIe2cDLvm5IqpeGFlMTD98HCqv7+GE54anRD/zbvGNYtOsXMhAuUTyasIhvj2KPfN\n"
-    " \tRbmmIBnqNUzidgkKb244eP794ZpMdkYwRAIgNVq8SYP7js0C/GAGMKVYXiCGUTIL7OKPSBLS     \n"
-    "\t7LTyrL4CIE+s4Tsn/FrrYj0nMEV1Mvf7PMRYCxtEERD3PG/etTJ3cBJAbwWWofHqg9IACoYV\n"
-    "\t +n9ulZHSVRajo55EkZYw0XUXDw8zcI4gD58suOSLZTG/dXtZp17huIyHgxHbR2YeYjQpCw==\t  \t";
+    "JAAAAAFxIe2cDLvm5IqpeGFlMTD98HCqv7+GE54anRD/zbvGNYtOsXMhAuUTyasIhvj2KPfN"
+    "RbmmIBnqNUzidgkKb244eP794ZpMdkYwRAIgNVq8SYP7js0C/GAGMKVYXiCGUTIL7OKPSBLS"
+    "7LTyrL4CIE+s4Tsn/FrrYj0nMEV1Mvf7PMRYCxtEERD3PG/etTJ3cBJAbwWWofHqg9IACoYV"
+    "+n9ulZHSVRajo55EkZYw0XUXDw8zcI4gD58suOSLZTG/dXtZp17huIyHgxHbR2YeYjQpCw==";
```
Comment (seelabs) [inline src/test/rpc/ServerInfo_test.cpp]: Why was this changed? It looks like the old test was testing for extra white space that this modified test no longer exercises.

File: src/test/jtx/envconfig.h [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+/// configuration for jtx::Env
+///
+/// @return unique_ptr to Config instance
+inline
+std::unique_ptr<Config>
+envconfig()
+{
+    auto p = std::make_unique<Config>();
+    setupConfigForUnitTests(*p);
+    return p;
```
Comment (seelabs) [inline src/test/jtx/envconfig.h]: How about changing `modfunc` so it takes the `unique_ptr` as a parameter and returns a `unique_ptr`? (we could also rename it to `f`). So the whole function would be: `return f(envconf(), std::forward<Args>(args)...);`

File: src/test/jtx/envconfig.h [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+/// @brief adjust the default configured server ports by a specified value
+///
+/// This is intended for use with envconfig, as in
+/// envconfig(portincrement, 5)
+///
+/// @param cfg config instance to be modified
+/// @param int amount by which to increment the existing server port
+/// values in the config
+void
+portincrement(Config&, int);
```
Comment (seelabs) [inline src/test/jtx/envconfig.h]: Nit: The name `portincrement` is neither camelCase nor snake_case. I know jtx doesn't always conform to the rippled style guild, but I think this name is out of bounds (same with the other multiple word names that run together).

File: src/test/rpc/ServerInfo_test.cpp (hunk 1 above)
Comment (mellery451) [inline src/test/rpc/ServerInfo_test.cpp]: the existing code was calling `Config::loadFromString` which calls `parseIniFile` which does trimming. Our new code path does none of this but instead manipulates the config using the validator() helper, thus these extra spaces/chars are not allowed. If we want direct tests of `loadFromString` and `parseIniFile` I'm all for it, but I don't think this test is the right place for that test.

File: src/test/jtx/envconfig.h (hunk 2 above)
Comment (mellery451) [inline src/test/jtx/envconfig.h]: `modfunc` taking `unique_ptr` implies transfer of ownership in and then back out? or would it take `unique_ptr const&`? In any event, I think the semantics of `modfunc` should be modify but not own.

File: src/test/jtx/envconfig.h (hunk 3 above)
Comment (mellery451) [inline src/test/jtx/envconfig.h]: looks like other jtx helpers are using snake, so I'll go with that.

File: src/test/jtx/envconfig.h (hunk 2 above)
Comment (seelabs) [inline src/test/jtx/envconfig.h]: Yes, I'd have it transfer ownership and then back out. All of these `mod` functions are meant to be used with `envconf` (if that wasn't the case, then passing the `unique_ptr` would be a bad interface). However, I don't see a scenario where they will be reused for anything other than `envconf`.

Pros:
* This would allow a function to create a completely new config if it desires
* The interface is clearer to me when modifications are returned rather than modifying a parameter (I dislike in/out parameters).

While I'll defend my POV, this is a minor issue. If you like the `in/out` parameter better it's your call.

File: src/test/app/Regression_test.cpp [hunk 4]
```diff
@@ -165,10 +165,11 @@ struct Regression_test : public beast::unit_test::suite
     {
         testcase("Autofilled fee should use the escalated fee");
         using namespace jtx;
-        Env env(*this, envconfig([](Config& cfg)
+        Env env(*this, envconfig([](std::unique_ptr<Config> cfg)
```
Comment (seelabs) [inline src/test/app/Regression_test.cpp]: We could make the params to these lambdas `auto`, for terseness. No change needed.

**Comment by mellery451** [comment]
this has conflicts with some other passed PRs - will rebase and resolve after the next beta release

**Comment by scottschurr** [comment]
Merged to develop as 80d9b0464a47af3252f1fafbaff32ba0cac195bb
